lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -248,13 +264,14 @@ def update(self, subsection, block_structure=None):
         if not PersistentGradesEnabledFlag.feature_enabled(self.course.id):
             return
 
-        block_structure = self._get_block_structure(block_structure)
-        subsection_grade = SubsectionGrade(subsection, self.course)
-        subsection_grade.compute(self.student, block_structure, self._scores_client, self._submissions_scores)
+        with persistence_safe_fallback():
```

Comment:
Hmm.. we have to consider the ramifications of doing this within `update`.  For the "read" methods, it makes sense to have "try-again" in case of race conditions.  However, how would we recover from a conflict in an update?  I'd rather fail the entire transaction and have the update happen again.